### DISCUSSION TITLE
"Diffusers use exact steps"?

### DISCUSSION THREAD

**Discussion by Silanda** [vladmandic/sdnext#2114] [General]
Am I misunderstanding this change, or does this now mean that rather than the steps used by the refiner/upscaler being determined by "Denoise start" or "Denoise strength", they now have to be set manually? That's how it seems to behave anyway, and while I can see the point of that as an option, this seems confusing.

What impact do "Denoise start" and "Denoise strength" have now then if they don't alter the number of steps run?

Or is it the case that it's not working as it's supposed to on my end?

**Comment by vladmandic** [comment]
cc @Disty0

**Comment by Disty0** [comment]
They still alter the steps. We are calculating the number of steps to get what we set to be run.

Denoise start tells diffusers to start the scheduler at this percent and steps will be run after that point.
Still working as expected but we are now calculating the necessary steps needed to get the exact steps we want to run after denoise start point.

Same with denoise strength. 
It still denoises at what we specified but we are now calculating the steps needed to run the exact steps we want.

So if you select 0.3 denoise strength and 20 steps, Diffusers will use 67 steps now. (20 // 0.3 +1)
And 67 * 0.3 denoise strength will result in 20 actual steps.

**Comment by Silanda** [comment]
Well, I'll admit I'm a bit confused. 

With SDXL, if I set sampling steps to 50 and refiner steps to 10, with 0.8 refiner start, the console shows it running 50 base steps and 10 refiner steps. If I set it to 50 and 50, it runs 50 and 50, and the time it takes seems appropriate for that. Changing refiner start doesn't seem to alter the number of steps that the console indicates are being run, but it definitely alters the image output. Is this the intended behaviour?

Before, both base and refiner steps would be set to 50, and the result would be 40 and 10 in the console.

**Reply by Disty0** [reply]
It is the intended behavior.

Base stops at %80 (0.8 you've set) of it's generation independently from the Refiner and we calculate that %80 to be 50 steps (or whatever you set).
Refiners starts at %80 (0.8 you've set) of it's generation independently from the Base and we calculate the remaining %20 to be 10 steps (or whatever you set).

Refiner/Denoise start just sets the point for scheduler start and stop.
At the start, scheduler sets denoise to be high and lowers it as it goes. 
So starting at a later point will use less denoise.

Matching base and refiner steps isn't really necessary since scheduler percentage is the important thing here.

**Reply by Silanda** [reply]
Okay, thanks. I think I get it.
